src/lava/proc/io/bridge/extractor.py

```diff
@@ -0,0 +1,179 @@
+        buffer is full and how the dst_port behaves when the buffer is empty
+        and not empty.
+    """
+    def __init__(self,
+                 shape: tuple[int, ...],
+                 buffer_size: ty.Optional[int] = 50,
+                 extractor_channel_config: ty.Optional[ChannelConfig] =
+                 ChannelConfig()) -> None:
+        super().__init__()
+
+        validate_shape(shape)
+        validate_buffer_size(buffer_size)
+        validate_channel_config(extractor_channel_config)
+
+        self._shape = shape
+
+        self._extractor_channel_config = extractor_channel_config
+
+        self._multi_processing = MultiProcessing()
+        self._multi_processing.start()
+
+        self._extractor_channel = \
+            PyPyChannel(message_infrastructure=self._multi_processing,
+                        src_name="src",
+                        dst_name="dst",
+                        shape=self._shape,
+                        dtype=float,
+                        size=buffer_size)
+        self._extractor_channel_dst_port = self._extractor_channel.dst_port
+        self._extractor_channel_dst_port.start()
+
+        self.proc_params["shape"] = shape
+        self.proc_params["extractor_channel_config"] = extractor_channel_config
+        self.proc_params["extractor_channel_src_port"] = \
+            self._extractor_channel.src_port
+
+        if self._extractor_channel_config.recv_buffer_empty == \
+                ChannelRecvBufferEmpty.BLOCKING:
+            self._recv_empty = recv_empty_blocking
+        elif self._extractor_channel_config.recv_buffer_empty == \
+                ChannelRecvBufferEmpty.NON_BLOCKING_ZEROS:
+            self._recv_empty = recv_empty_non_blocking_zeros
+
+        if self._extractor_channel_config.recv_buffer_not_empty == \
+                ChannelRecvBufferNotEmpty.FIFO:
+            self._recv_not_empty = recv_not_empty_fifo
+        elif self._extractor_channel_config.recv_buffer_not_empty == \
+                ChannelRecvBufferNotEmpty.ACCUMULATE:
+            self._recv_not_empty = recv_not_empty_accumulate
+
+        self.in_port = InPort(shape=shape)
+
+    def recv_data(self) -> np.ndarray:
+        """Receive data from the ProcessModel.
+
+        The data is received from extractor_channel.dst_port.
+
+        Returns
+        ----------
+        data : np.ndarray
+            Data received.
+        """
+        elements_in_buffer = \
+            self._extractor_channel_dst_port._queue._qsize()
+
+        if elements_in_buffer == 0:
+            data = self._recv_empty(
+                dst_port=self._extractor_channel_dst_port,
+                zeros=np.zeros(self._shape))
+        else:
+            data = self._recv_not_empty(
+                dst_port=self._extractor_channel_dst_port,
+                zeros=np.zeros(self._shape),
+                elements_in_buffer=elements_in_buffer)
+
+        return data
+
+    def __del__(self) -> None:
+        super().__del__()
```

Comment:
If you follow the comment under __init__, you should be able to get rid of this, which is good because this will leak whenever a notebook is killed/restarted.

Reply:
See comment under init.

Reply:
Resolved. We will merge working version and fix issues with MultiProcessing separately.